### PR TITLE
[0.3.1] Create from existing agent guide

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

Added a tutorial/guide on how to create and register a service with an existing agent on-chain.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/guides/create_service_existing_agent.md [hunk 1]
```diff
@@ -1 +1,212 @@
-This tutorial will be added soon.
\ No newline at end of file
+The typical use case that this guide intends to illustrate is where a service owner wants to use already developed agents to run its own service, possibly customizing it through the service parameters. For example, an oracle agent service with existing agents, where the service owner specifies
```
Comment (Karrenbelt) [inline docs/guides/create_service_existing_agent.md:1]: ...where a service owner wants to use *existing* agents to run their own service *with custom parameters*. For example, an oracle *service customized to serve data of their interest*.

File: docs/guides/create_service_existing_agent.md [hunk 2]
```diff
@@ -1 +1,212 @@
...
+
+    !!!info "The structure of the service definition file"
+        As its name suggests, the service definition file `service.yaml` is where
+        the parameters of the agent service are defined, including the particular agent that composes the service. Here is an example of a `service.yaml` file:
+        ```yaml
+        name: hello_world_2_service
+        author: your_name
+        version: 0.1.0
+        description: This is the Hello World 2 service.
+        aea_version: '>=1.0.0, <2.0.0'
```
Comment (Karrenbelt) [inline docs/guides/create_service_existing_agent.md:50]: can you confirm this version range is supported?

**Review by Karrenbelt — COMMENTED** [review]
what I'm missing is an explanation - or rather showcasing - of how an agent oracle service can be customized as you start out with saying you'll do.

File: docs/guides/create_service_existing_agent.md (hunk 2 above)
Comment (jmoreira-valory) [reply docs/guides/create_service_existing_agent.md:50]: This is copied from hello-world 1, so I assume so.

**Comment by jmoreira-valory** [comment]
> what I'm missing is an explanation - or rather showcasing - of how an agent oracle service can be customized as you start out with saying you'll do.

I totally agree. That's why I suggested in the devx issues document to add the message "hello world" as a service parameter that can be changed later easily in "hello world 2" in this tutorial.
I think using more complex service like the oracle, will divert too much the attention to the focus of the tutorial into many details which is precisely what we want to avoid: give precise indications and remove the details of what it is not relevant at this point.... (IMO) If there is an agreement to use the oracle, this tutorial can be modified.

File: docs/guides/create_service_existing_agent.md (hunk 1 above)
Comment (Karrenbelt) [reply docs/guides/create_service_existing_agent.md:1]: parameters -> configuration

File: docs/guides/create_service_existing_agent.md [hunk 3]
```diff
@@ -1 +1,212 @@
...
+
+
+4. **Publish the service on the [IPFS](https://ipfs.io/).** This will make the service available for other developers to fetch it.
+
+    !!! warning "Important"
+        Before publishing any service or agent, make sure that all the files and folders no longer needed are deleted. In this example,
+            ```bash
+            rm keys.json
+            rm -rf abci_build
+            ```
```
Comment (angrybayblade) [inline docs/guides/create_service_existing_agent.md:158]: We need to make sure publish command takes care of this. https://github.com/valory-xyz/open-autonomy/issues/1315

File: docs/guides/create_service_existing_agent.md [hunk 4]
```diff
@@ -1 +1,212 @@
...
+    ```
+    valory/hello_world:0.1.0:bafybeicealdcbxjdejskntddizntwqmlpyxa2ujaxnw2cgy73x3swldwcq
+    ```
+
+    If you want, you can browse the agent contents stored in the IPFS [here](https://gateway.autonolas.tech/ipfs/bafybeicealdcbxjdejskntddizntwqmlpyxa2ujaxnw2cgy73x3swldwcq/hello_world/).
+
+    !!!note
+        Future releases of the {{open_autonomy}} framework will provide convenient commands to browse and discover existing registered components. For now, we assume that we already know the IPFS hash of the agent.
+
+2. **Create the service definition.** Create an empty folder with the service name (e.g., `hello_world_2_service`). Inside that folder, create a `README.md` file, where you can write a description of the service, and a service definition file `service.yaml`, where the agent IPFS hash must be specified:
```
Comment (angrybayblade) [inline docs/guides/create_service_existing_agent.md:33]: Why not just `hello_world`?

Comment (jmoreira-valory) [reply docs/guides/create_service_existing_agent.md:32]: not to confuse the reader with the existing hello_woprld

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
You reintroduced two submodules. Please remove

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**